### ISSUE TITLE
Improve SCF environment initializations

### ISSUE THREAD

**Issue by lelemmen** [GQCG/GQCP#885]
I've been thinking about the initialization of the SCF environments recently. Currently, we offer three named ways to initialize and set up SCF environments:

https://github.com/GQCG/GQCP/blob/3fa1179ad40829d46d642c670e0f400e96942467/gqcp/include/QCMethod/HF/GHF/GHFSCFEnvironment.hpp#L90-L108

https://github.com/GQCG/GQCP/blob/3fa1179ad40829d46d642c670e0f400e96942467/gqcp/include/QCMethod/HF/GHF/GHFSCFEnvironment.hpp#L119-L140

https://github.com/GQCG/GQCP/blob/3fa1179ad40829d46d642c670e0f400e96942467/gqcp/include/QCMethod/HF/GHF/GHFSCFEnvironment.hpp#L153-L164

However, I'm not so sure that these APIs have been correctly identified, even though they work just fine. Since all we're doing is essentially providing an initial guess for the coefficient matrix, I think it would be better to move these APIs **to the level of `Transformation`**. Concerning the three APIs mentioned above, I would rewrite them as:

```cpp
GTransformation::CoreDiagonalization(const size_t N, const GSQHamiltonian<Scalar>& sq_hamiltonian, const ScalarGSQOneElectronOperator<Scalar>& S);
```

```cpp
SquareMatrix::complexify(const double x = 0.1);
```

The final example is obsolete, since `Eigen` offers `MatrixBase::unaryExpr()` that we can use to apply a unary function to an `Eigen` matrix object.

@xdvriend @dariatols What are your thoughts?

**Comment by xdvriend** [comment]
I see your point! 
Would you then move towards one way of setting up an SCFEnvironment where you need to give the initial guess as an argument? e.g.: 
```c++
const auto guess = GTransformation::CoreDiagonalization(const size_t N, const GSQHamiltonian<Scalar>& sq_hamiltonian, const ScalarGSQOneElectronOperator<Scalar>& S);
const auto environment = GHFSCFEnvironment<Scalar>(guess.matrix());
```
Or do you have something else in mind for that?

**Comment by lelemmen** [comment]
Indeed, that kind of initialization for the environment would be the direction I would propose to go in.

**Comment by xdvriend** [comment]
I think that's very clear! It would also allow us to provide more options to initialise an `SCFEnvironment` in a much cleaner way.

**Comment by dariatols** [comment]
Indeed, this makes the initialization of an `SCFEnvironment` object more intuitive in my opinion!
